File: src/main/java/com/harshit/pharmacy/payment/service/PaymentService.java
package com.harshit.pharmacy.payment.service;

import com.harshit.pharmacy.order.entity.Order;
import com.harshit.pharmacy.payment.dto.PaymentRequest;
import com.harshit.pharmacy.payment.dto.PaymentResponse;
import com.harshit.pharmacy.payment.entity.Payment;

import java.util.List;

public interface PaymentService {

    PaymentResponse processOnlinePayment(PaymentRequest request);

    void collectCodPayment(Integer orderId);

    List<PaymentResponse> getMyPayments();

    PaymentResponse getPayment(Integer paymentId);

    void createCodPayment(Order order);

    void updatePaymentMethod(Order order, String paymentMethod);

}

File: pharmacy-frontend/src/services/paymentService.js
import api from "../api/api";

export async function getMyPayments() {
    const response = await api.get("/payments");
    return response.data;
}

export async function getPayment(paymentId) {
    const response = await api.get(`/payments/${paymentId}`);
    return response.data;
}

export async function processOnlinePayment(orderId, paymentMethod) {
    const response = await api.post("/payments/online", {
        orderId,
        paymentMethod
    });

    return response.data;
}